central/networkgraph/entity/datastore/datastore_impl_test.go

```diff
@@ -633,3 +640,1023 @@ func (suite *NetworkEntityDataStoreTestSuite) expectPushExternalNetworkEntitiesT
+		},
+		"Cluster full access cannot get global entity": {
+			contextName:   sacTestUtils.Cluster1ReadWriteCtx,
+			entityID:      entity1ID.String(),
+			expectedFound: false,
+		},
+		"Cluster full access can get cluster entity": {
+			contextName:    sacTestUtils.Cluster1ReadWriteCtx,
+			entityID:       entity2ID.String(),
+			expectedFound:  true,
+			expectedEntity: clusterEntity,
+		},
+		"Cluster full access cannot get missing entity": {
+			contextName:   sacTestUtils.Cluster1ReadWriteCtx,
+			entityID:      entity3ID.String(),
+			expectedFound: false,
+		},
+		"Cluster partial access cannot get global entity": {
+			contextName:   sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			entityID:      entity1ID.String(),
+			expectedFound: false,
+		},
+		"Cluster partial access cannot get cluster entity": {
+			contextName:   sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			entityID:      entity2ID.String(),
+			expectedFound: false,
+		},
+		"Cluster partial access cannot get missing entity": {
+			contextName:   sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			entityID:      entity3ID.String(),
+			expectedFound: false,
+		},
+		"Other cluster access cannot get global entity": {
+			contextName:   sacTestUtils.Cluster2ReadWriteCtx,
+			entityID:      entity1ID.String(),
+			expectedFound: false,
+		},
+		"Other cluster access cannot get cluster entity": {
+			contextName:   sacTestUtils.Cluster2ReadWriteCtx,
+			entityID:      entity2ID.String(),
+			expectedFound: false,
+		},
+		"Other cluster access cannot get missing entity": {
+			contextName:   sacTestUtils.Cluster2ReadWriteCtx,
+			entityID:      entity3ID.String(),
+			expectedFound: false,
+		},
+	} {
+		s.Run(name, func() {
+			ctx := testContexts[tc.contextName]
+			entity, found, testErr := s.ds.GetEntity(ctx, tc.entityID)
+			s.NoError(testErr)
+			if tc.expectedFound {
+				s.True(found)
+				protoassert.Equal(s.T(), tc.expectedEntity, entity)
+			} else {
+				s.False(found)
+				s.Nil(entity)
+			}
+		})
+	}
+}
+
+func (s *NetworkEntityDataStoreTestSuite) TestGetEntityByQuerySAC() {
+	// Note: The test here reflects the observed behaviour,
+	// which is not the expected one from a pure scoped access control perspective
+	// (entities that are not linked to the requester scope should be filtered out,
+	// like global entities for users that have access scope restricted to a cluster
+	// or cluster entities for clusters that are not in the access scope of the user).
+	entityIDs, entities := s.setupSACReadSingleTest()
+	entity1ID := entityIDs[0]
+	entity2ID := entityIDs[1]
+	entity3ID := entityIDs[2]
+	globalEntity := entities[0]
+	clusterEntity := entities[1]
+	testContexts := sacTestUtils.GetNamespaceScopedTestContexts(context.Background(), s.T(), resources.NetworkGraph)
+
+	allEntities := []*storage.NetworkEntity{globalEntity, clusterEntity}
+	// clusterEntities := []*storage.NetworkEntity{clusterEntity}
+	// noEntities := []*storage.NetworkEntity{}
```

Comment:
Why it's commented?

Reply:
The point here is to highlight the difference between the test behaviour and the expected one.
The current implementation of GetEntityByQuery does not perform much access control checks. The commented test code or test descriptions highlight what the actual function behaviour could or should be.